src/components/draw/DrawDirective.js

```diff
@@ -464,8 +464,37 @@ goog.require('ga_permalink');
             }));
 
             unDrawEvts.push(draw.on('drawend', function(evt) {
+              // Unregister the change event
+              ol.Observable.unByKey(deregFeatureChange);
+              deactivateTool(lastActiveTool);
+              scope.$applyAsync();
+
               var featureToAdd = evt.feature;
               var geom = featureToAdd.getGeometry();
+
+              // According to #3319, it seems LineString can be created with one
+              // point (or with an array of exact same points). If it's the case
+              // we ignore it.
+              if (geom instanceof ol.geom.Polygon) {
+                var hasUniqueCoords = true;
+                var coords;
+                geom.getCoordinates()[0].forEach(function(item) {
```

Comment:
Similar function is in kml service.

Maybe we can add a kml validation service and put those validations there. Then use it in this directive and in the kml service.

Thoughts?